src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -719,5 +726,288 @@ .maxstack  2
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.StaticData.get", @$"{{
+  // Code size       29 (0x1d)
+  .maxstack  4
+  IL_0000:  ldc.i4.3
+  IL_0001:  newarr     ""char""
+  IL_0006:  dup
+  IL_0007:  ldc.i4.0
+  IL_0008:  ldc.i4.s   97
+  IL_000a:  stelem.i2
+  IL_000b:  dup
+  IL_000c:  ldc.i4.1
+  IL_000d:  ldc.i4.s   98
+  IL_000f:  stelem.i2
+  IL_0010:  dup
+  IL_0011:  ldc.i4.2
+  IL_0012:  ldc.i4.s   99
+  IL_0014:  stelem.i2
+  IL_0015:  ldc.i4.{start}
+  IL_0016:  ldc.i4.{length}
+  IL_0017:  newobj     ""System.ReadOnlySpan<char>..ctor(char[], int, int)""
+  IL_001c:  ret
+}}");
+        }
+
+        [Fact]
+        public void CreateSpanIsNotUsedForNonComplainTypes()
+        {
+            var csharp = RuntimeHelpersCreateSpan + $@"
+public struct SingleByteField
+{{
+    public byte Value;
+}}
+
+public class Test
+{{
+    public static System.ReadOnlySpan<SingleByteField> StaticData => new SingleByteField[] {{ default, default, default }};
+}}
+";
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, options: TestOptions.UnsafeReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.StaticData.get", @$"{{
+  // Code size       12 (0xc)
+  .maxstack  1
+  IL_0000:  ldc.i4.3
+  IL_0001:  newarr     ""SingleByteField""
+  IL_0006:  call       ""System.ReadOnlySpan<SingleByteField> System.ReadOnlySpan<SingleByteField>.op_Implicit(SingleByteField[])""
+  IL_000b:  ret
+}}");
+        }
+
+        [Theory]
+        [InlineData("byte", 1)]
+        [InlineData("short", 2)]
+        [InlineData("int", 4)]
+        [InlineData("long", 8)]
+        public void Alignment_FieldsAreAlignedAndPackedAccordingToType(string typeName, int expectedAlignment)
+        {
+            string csharp = RuntimeHelpersCreateSpan + $@"
+public class Test
+{{
+    public static System.ReadOnlySpan<{typeName}> Data => new {typeName}[] {{ 1, 2, 3 }};
+}}";
+
+            var compilation = CreateCompilation(csharp, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.UnsafeReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyTypeIL("<PrivateImplementationDetails>", il =>
+            {
+                Assert.Contains($".pack {expectedAlignment}", il);
+
+                Match m = Regex.Match(il, @"\.data cil I_([0-9A-F]*) = bytearray");
+                Assert.True(m.Success, $"Expected regex to match in {il}");
+                Assert.True(long.TryParse(m.Groups[1].Value, NumberStyles.HexNumber, null, out long rva), $"Expected {m.Value} to parse as hex long.");
+                Assert.True(rva % expectedAlignment == 0, $"Expected RVA {rva:X8} to be {expectedAlignment}-byte aligned.");
+            });
+        }
```

Comment:
Code and tests look good to me, but we should still chase that SRM version question. Let's sync with Jared on this tomorrow.